lightning/src/ln/channelmanager.rs

```diff
@@ -4291,12 +4290,16 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 						} else { None };
 
 						let mut pending_events = self.pending_events.lock().unwrap();
+
+						let channel_id = if fee_earned_msat.is_some() { Some(prev_outpoint.to_channel_id()) } else { None };
```

Comment:
Heh, oops, I didn't mean "set it to None if the channel is gone" I meant "dont test for it if the `fee_earned_msat` field is None in tests", we should include it anyways because we have it and users may care.